src/AppInstallerCLICore/Workflows/RepairFlow.cpp

```diff
@@ -0,0 +1,554 @@
+        }
+
+        IPackageVersion::Metadata packageMetadata = installedPackageVersion->GetMetadata();
+
+        auto noModifyItr = packageMetadata.find(PackageVersionMetadata::NoModify);
+        std::string noModifyARPFlag = noModifyItr != packageMetadata.end() ? noModifyItr->second : std::string();
+
+        auto noRepairItr = packageMetadata.find(PackageVersionMetadata::NoRepair);
+        std::string noRepairARPFlag = noRepairItr != packageMetadata.end() ? noRepairItr->second : std::string();
+
+        if (Utility::IsDwordFlagSet(noModifyARPFlag) || Utility::IsDwordFlagSet(noRepairARPFlag))
+        {
+            context.Reporter.Error() << Resource::String::RepairOperationNotSupported << std::endl;
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_REPAIR_NOT_SUPPORTED);
+        }
+    }
+
+
+    /// <summary>
+    /// This function performs a preliminary check on the available matching package by reading its manifest entries for repair behavior to determine the type of repair operation and repair switch are applicable
+    /// </summary>
+    /// <param name="context">Execution context object</param>
+    void ApplicabilityCheckForAvailablePackage(Execution::Context& context)
+    {
+        // Selected Installer repair applicability check
+        auto const& installerType = context.Get<Execution::Data::Installer>()->EffectiveInstallerType();
+        auto const& repairBehavior = context.Get<Execution::Data::Installer>()->RepairBehavior;
+
+        if (installerType == InstallerTypeEnum::Portable || installerType == InstallerTypeEnum::Unknown)
+        {
+            context.Reporter.Error() << Resource::String::RepairOperationNotSupported << std::endl;
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_REPAIR_NOT_SUPPORTED);
+        }
+
+        // For other installer types, we can skip this check as we expect standard repair support from platform.
+        if (installerType == InstallerTypeEnum::Burn
+            || installerType == InstallerTypeEnum::Inno
+            || installerType == InstallerTypeEnum::Nullsoft
+            || installerType == InstallerTypeEnum::Exe)
+        {
+            if (repairBehavior == RepairBehaviorEnum::Unknown)
+            {
+                context.Reporter.Error() << Resource::String::NoRepairInfoFound << std::endl;
+                AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_NO_REPAIR_INFO_FOUND);
+            }
+            else if (repairBehavior == RepairBehaviorEnum::Installer)
+            {
+                context <<
+                    Workflow::EnsureSupportForDownload <<
+                    Workflow::EnsureSupportForInstall;
+            }
+        }
+    }
+
+    void RepairApplicabilityCheck(Execution::Context& context)
+    {
+        context <<
+            ApplicabilityCheckForInstalledPackage <<
+            ApplicabilityCheckForAvailablePackage;
+    }
+
+    void ExecuteRepair(Execution::Context& context)

```

Comment:
Also add a TODO: comment somewhere that current repair flow requires an available installer mapped. In the future, we should do some refactoring to enable repair of msix/msi without requiring a mapped available installer.